index.js

```diff
@@ -184,9 +189,18 @@ class LatticeKeyring extends EventEmitter {
         encodingType: SDK.Constants.SIGNING.ENCODINGS.EVM,
         signerPath,
       };
-      const supportsDecoderRecursion = fwVersion.major > 0 || fwVersion.minor >=16;
-      // Check if we can decode the calldata
-      const { def } = await SDK.Utils.fetchCalldataDecoder(tx.data, tx.to, chainId, supportsDecoderRecursion);
+      let def;
+      const cachedDef = getCachedDef(tx, this.requestCache)
+
+      if (cachedDef) {
+        def = cachedDef
+      } else {
+        const supportsDecoderRecursion = fwVersion.major > 0 || fwVersion.minor >=16;
```

Comment:
Oooh I just realized... what happens if someone switches Lattices and ends up with an older fw version? Do we need to nuke `this.requestCache` when creds get cleared?

Maybe not worrying about this complexity if we can just tell people to update their firmware. Thoughts?

Reply:
I added a reset of the cache to `_resetDefaults()` which should cover us in that use case. [4eb1101](https://github.com/GridPlus/eth-lattice-keyring/commit/4eb1101cd87239084ac1fcca4a2d45b376dc651d)